main.tf

```diff
@@ -90,6 +90,28 @@ resource "equinix_metal_spot_market_request" "join_spot_request" {
   }
 }
 
+data "equinix_metal_device" "seed_device" {
+   device_id = data.equinix_metal_spot_market_request.seed_req.0.device_ids[0]
+}
+
+resource "equinix_metal_vlan" "vlans" {
+  count = var.num_of_vlans
+  project_id = data.equinix_metal_project.project.project_id
+  facility = var.facility
+}
+
+resource "equinix_metal_device_network_type" "seed_nw_type" {
+  device_id = data.equinix_metal_device.seed_device.id
+  type      = "hybrid"
+}
+
+resource "equinix_metal_port_vlan_attachment" "vlan_attach" {
+   count = var.num_of_vlans >=1 ? 1 : 0
```

Comment:
I think this is causing only the first VLAN to be attached and not all. Is that intentional? Think we want to attach all VLANs.

Reply:
And I think we want to attach all VLANs to all hosts, not just the seed.

Reply:
Ok, so the understanding was correct, but I wasn't able to find a way to do it. I think I have a few questions:

1  - How do we define which VLAN should connect to which hosts?
2 - If we are creating 2 VLANs and have 2 hosts, should we just attach to each?
3 - What If we have 2 hosts but create 3 or more VLANs?

Reply:
1 - connect all VLANs to all hosts
2 - yes, 4 total attachments.
3 - then will have 4 or more attachments

It might be a little tricky, but I think we can do this by setting `count` to var.num_of_vlans * (var.node_count + 1).  For `device_id`, you'll need logic to determine if it's seed or join and same for the `vlan_vnid`.